### PR TITLE
lig 3050 simclr benchmark improvements

### PR DESCRIPTION
**Pull request by guarin**
## Changes

* Encode all views at once. This is how it is implemented in the original paper. It should slightly improve training as the model  cannot use batch norm statistics to differentiate between embeddings from view 1 and 2. Not sure how much it improves though, didn't benchmark it.
* Use square root learning rate scaling. This is recommended by the paper for smaller batch sizes.
* Use 0.9 momentum. Forgot to add this previously. This is why the results are now much better.

## How was it tested?

Ran the benchmark again. We get now 63.2% linear eval top1 accuracy, compared to the paper which reports 62.8% for the same settings. Full results:
```
Imagenet
	max val_online_cls_top1: 	0.6309599876403809
	max val_online_cls_top5: 	0.8507199883460999
	knn val_top1: 			0.44929999113082886
	knn val_top5: 			0.7418799996376038
	max linear val_top1: 		0.6319800019264221
	max linear val_top5: 		0.8527600169181824

Imagenet100
	max val_online_cls_top1: 	0.6898000240325928
	max val_online_cls_top5: 	0.8980000019073486
	knn val_top1: 			0.5752000212669373
	knn val_top5: 			0.8485999703407288
	max linear val_top1: 		0.7233999967575073
	max linear val_top5: 		0.9165999889373779
	max finetune val_top1: 		0.7954000234603882
	max finetune val_top5: 		0.949400007724762

Imagenette
	max val_online_cls_top1: 	0.8166072368621826
	max val_online_cls_top5: 	0.9801324605941772
	knn val_top1: 			0.7763627171516418
	knn val_top5: 			0.9793683290481567
	max linear val_top1: 		0.8494651317596436
	max linear val_top5: 		0.9857361316680908
	max finetune val_top1: 		0.6314314603805542
	max finetune val_top5: 		0.943708598613739
```

Training was 100 epochs with batch size 256 (=2x128). This corresponds to the first entry in this table:
Screen Shot 2023-05-08 at 13 51 04

Will make a follow-up PR to add the results to readme/docs.

### CODE REVIEW

File: benchmarks/imagenet/resnet50/knn_eval.py [hunk 1]
```diff
@@ -75,4 +75,4 @@ def knn_eval(
         val_dataloaders=val_dataloader,
     )
     for metric in ["val_top1", "val_top5"]:
-        print(f"{metric}: {metric_callback.val_metrics[metric]}")
+        print(f"knn {metric}: {max(metric_callback.val_metrics[metric])}")
```
Comment (guarin) [inline benchmarks/imagenet/resnet50/knn_eval.py:78]: This is just cleanup

**Review by philippmwirth — APPROVED** [review]
